### PR TITLE
Oni2: Update Python and Update License

### PR DESCRIPTION
**Pull request by Krutonium**
###### Description of changes

  - Updated the License from the Vendor License (that only applies to builds that they distribute) to the MIT license that the source code is under
  - Updated the code to use Python38, as it was previously targeting Python2 and, and immediately failing as it wanted at least Python37
  
  Current Problem, that I hope someone could help me fix, it currently fails to compile with an error in `signals_nat.c` and I don't know enough C to fix it.
[oni2.log](https://github.com/NixOS/nixpkgs/files/11069629/oni2.log)

As a result this is currently a non building PR in the hope of assistance so it can be merged. (As far as I can tell, this has never been able to build since the day it was added, or the package is pulling in dependencies from outside Nix that have updated since.)

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by teto** [comment]
I thought oni2 was dead  ?

**Comment by Krutonium** [comment]
> I thought oni2 was dead  ?

It might be as a project? But as far as I can tell it's never worked on NixOS, and a friend of mine is trying to use it on NixOS, and there is no reason to not see about making it work - just because the project is dead doesn't mean there isn't still users of it.

**Comment by teto** [comment]
> As a result this is currently a non building PR in the hope of assistance so it can be merged. (As far as I can tell, this has never been able to build since the day it was added, or the package is pulling in dependencies from outside Nix that have updated since.)

this definitely built at some point in time/was working as I remember testing it (as I asked for it to be packaged). I am just mentioning that the project is dead because you request outside help. I think maintainers time is best spent on alive projects (like neovide/gonvim)but I ofc I would be happy if we had a running oni2.

File: pkgs/applications/editors/oni2/default.nix [hunk 1]
```diff
@@ -4,9 +4,9 @@ let mkOni2 = callPackage ./common.nix { };
 in mkOni2 rec {
   variant = "oni2";
   license = {
```
Comment (gardspirito) [inline pkgs/applications/editors/oni2/default.nix:6]:
```suggestion
  license = lib.licenses.mit;
```

File: pkgs/applications/editors/oni2/default.nix [hunk 2]
```diff
@@ -4,9 +4,9 @@ let mkOni2 = callPackage ./common.nix { };
...
-    fullName = "Outrun Labs End User License Agreement";
-    url = "https://github.com/onivim/oni2/blob/master/Outrun-Labs-EULA-v1.1.md";
-    free = false;
+    fullName = "MIT";
+    url = "https://github.com/onivim/oni2/blob/master/LICENSE.md";
+    free = true;
   };
```
Comment (gardspirito) [inline pkgs/applications/editors/oni2/default.nix:10]:
```suggestion
```

**Review by luna-spirito — CHANGES_REQUESTED** [review]
Hey. Thanks for that! Sorry for not updating it, my fault.

**Comment by gardspirito** [comment]
Regarding the build issue. Esy.sh doesn't have a proper Nix integration, so this derivation was made inherently impure, and I'm not surprised of it being broken.

There are two points of failure:
1) `oni2-fetch-deps`. This derivation performs all the impure part and unsafely fetches external dependencies. If the `outputHash` is the same as the hard-coded one, then it succeeds and that means that the source code and all dependencies are still the same and I didn't mess up at some point.
2) `oni2`. It performs almost all the building, and if it's broken, then it's some issue with updates made to Nixpkgs. 

I'll have a look at this today or tomorrow. Thanks for raising this issue.

**Comment by gardspirito** [comment]
I forgot to write here.

So, after some digging, it becomes evident that the project is really outdated. One of the dependencies of Oni2, skia, depended on Python 2 in the past. This dependency was removed over 3 years ago, making esy work with both Python 2 and Python 3, but Oni2 didn't update skia, so now it's just broken.

This is fixable, but even then there were some updates to the building system in Nixpkgs which are incompatible with skia because how old it is. Maybe I have mistaken, but that's just the impression I got.

Build failures are fixable (I hope), but I don't have a feeling this is worth fixing and worth being merged into Nixpkgs.

**Comment by Krutonium** [comment]
I'm going to close and give up on this then. Thanks for having a look :+1:

**Pull request closed without merging**